File: Isomorphic_Strings/isomorphic_str.dart
class Solution {
  bool isIsomorphic(String s, String t) {
    final firstMap = {};
    final sencondMap = {};

    for (var i = 0; i < s.length; i++) {
      if (firstMap.containsKey(s[i]) && firstMap[s[i]] != t[i]) {
        return false;
      }
      firstMap[s[i]] = t[i];
    }

    for (var i = 0; i < s.length; i++) {
      if (sencondMap.containsKey(t[i]) && sencondMap[t[i]] != s[i]) {
        return false;
      }
      sencondMap[t[i]] = s[i];
    }

    return true;
  }
}

File: Isomorphic_Strings/isomorphic_str.js
/**
 * @param {string} s - first string
 * @param {string} t - second string
 * @returns {boolean} returns whether both strings are isomorphic
 */
var isIsomorphic = function (s, t) {
	const firstMap = {};
	const secondMap = {};

	for (let i = 0; i < s.length; i++) {
		if (s[i] in firstMap && firstMap[s[i]] != t[i]) {
			return false;
		}

		firstMap[s[i]] = t[i];

		if (t[i] in secondMap && secondMap[t[i]] != s[i]) {
			return false;
		}

		secondMap[t[i]] = s[i];
	}

	return true;
};
